app/assets/javascripts/components/revisions/revisions_handler.jsx

```diff
@@ -54,7 +76,7 @@ const RevisionHandler = createReactClass({
           </div>
         </div>
         <RevisionList
-          revisions={this.props.revisions}
+          revisions={(this.state.courseSpecificBool) ? this.props.revisions.filter((revision) => { return this.props.course_specific_revision_ids.includes(revision.id); }) : this.props.revisions}
```

Comment:
I don't like doing the filtering here. This should probably be implemented as a `selector`; see `ArticlesHandler` for an example, where the state of the currently selected filters is used to do the filtering in `mapStateToProps`.

In addition to following the pattern of filtering elsewhere in the system, this approach also provides automatic caching of the filtered results; otherwise, every time one of the props for this component changes (which may happen a lot with wikidataLabels, in particular, as they get fetched after the rest of the data has been received) that `filter()` call will run again. Probably not a huge performance issue, but it might matter.

Reply:
Done in new PR #4002